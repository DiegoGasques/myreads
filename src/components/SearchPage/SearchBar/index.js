import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";

import "./styles.css";

function SearchBar({ handleSearch }) {
  return (
    <div className="search-books-bar">
      <Link to="/" className="close-search">
        Close
      </Link>
      <div className="search-books-input-wrapper">
        <input
          type="text"
          placeholder="Search by title or author"
          onKeyUp={e => handleSearch(e.target.value)}
        />
      </div>
    </div>
  );
}

SearchBar.propTypes = {
  handleSearch: PropTypes.func.isRequired
};

export default SearchBar;
