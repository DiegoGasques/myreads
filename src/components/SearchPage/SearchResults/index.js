import React from "react";
import PropTypes from "prop-types";

import List from "../../List";
import BookCard from "../../BookCard";
import { AppContext } from "../../App";

import "./styles.css";

function SearchResults({ results }) {
  return (
    <div className="search-books-results">
      <AppContext.Consumer>
        {context => (
          <List>
            {results.map(b => {
              return (
                <li key={`${b.id}`} className="list-item">
                  <BookCard book={b} handleUpdate={context.updateBookStatus} />
                </li>
              );
            })}
          </List>
        )}
      </AppContext.Consumer>
    </div>
  );
}

SearchResults.propTypes = {
  results: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default SearchResults;
