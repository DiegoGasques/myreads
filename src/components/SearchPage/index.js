import React, { Component } from "react";

import SearchBar from "./SearchBar";
import SearchResults from "./SearchResults";
import * as BooksAPI from "../../utils/BooksAPI";
import "./styles.css";

class SearchPage extends Component {
  state = {
    results: []
  };

  search = query => {
    if (query) {
      BooksAPI.search(query)
        .then(results => this.setState({ results }))
        .catch(e => console.log(e));
    } else {
      this.setState({ results: [] });
    }
  };

  render() {
    return (
      <div className="search-books">
        <SearchBar handleSearch={this.search} />
        <SearchResults results={this.state.results} />
      </div>
    );
  }
}

export default SearchPage;
