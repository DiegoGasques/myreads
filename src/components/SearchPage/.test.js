import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import SearchPage from "./index";

it("Should match SearchPage component snapshot", () => {
  const wrapper = shallow(<SearchPage />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});
