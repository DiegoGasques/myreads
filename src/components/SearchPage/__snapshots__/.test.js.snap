// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match SearchPage component snapshot 1`] = `
<div
  className="search-books"
>
  <SearchBar
    handleSearch={[Function]}
  />
  <SearchResults
    results={Array []}
  />
</div>
`;
